Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp

```diff
@@ -961,7 +961,7 @@ namespace AZ::Render
     void AtomActorDebugDraw::UpdateActorInstance(EMotionFX::ActorInstance* actorInstance, float deltaTime)
     {
         // Find the corresponding trajectory trace path for the given actor instance
-        TrajectoryTracePath* trajectoryPath = FindTrajectoryPath(actorInstance);
+        auto trajectoryPath = FindTrajectoryPath(actorInstance);
```

Comment:
There's not really any shared ownership of this object going on. In both the `UpdateActorInstance` and the `RenderTrajectoryPath` methods, the trajectory path is only viewed, they don't need to be involved in the ownership. That's because the `AtomActorDebugDraw` instance itself owns the `TrajectoryPath`, and the `AtomActorDebugDraw` object won't be destroyed in the middle of one of its methods running.

I think this is a great find, but we should use `unique_ptr` instead, because we don't need shared ownership. Make the `m_trajectoryTracePaths` member a `AZStd::vector<AZStd::unique_ptr<TrajectoryTracePath>>`, and have `FindTrajectoryPath` return a raw pointer (`TrajectoryTracePath*`).

Reply:
done